Rename quickstart as tutorial

BillWagner opened the pull request:
Fixes #8359 

Update the "quickstarts" to use the more popular search term "tutorial". For a further classification in our set of tutorials these are referred to as "introduction to C#" tutorials.

BillWagner committed:
move quickstarts

They are now in tutorials/intro-to-csharp

BillWagner committed:
configure redirects

BillWagner committed:
update all internal links

Updated some of the accompanying text as well.

BillWagner committed:
update TOCs

This also combined some of the nested TOCs to increase visibility.

BillWagner committed:
update text of all introductory tutorials to use tutorial

The term quickstart has been removed from the "introduction to C# tutorials"

BillWagner commented:
@dend This PR includes redirections that move some of the YML files for interactive tutorials.  That doesn't work with the build system. What's the fix?

BillWagner committed:
fix build errors

The redirection engine only supports .md files. So, even though the files being moved are .yml source, the redirection file needs to use the .md extension.

/cc @dend who provided the work around.

BillWagner committed:
update link paths for build errors.

Both relative links and code includes have been updated.

BillWagner committed:
missed one build error

pkulikov commented on .openpublishing.redirection.json:
```diff
@@ -2033,6 +2033,66 @@
+        },
+        {
+            "source_path": "docs/csharp/quick-starts/arrays-and-collections.md",
+            "redirect_url": "docs/csharp/tutorials/intro-to-csharp/arrays-and-collections",
```
typos here and in all other entries of the "redirect_url": `docs` -> `/dotnet` (note the slash to start with)

Thraka replied:
great catch!

BillWagner replied:
That is a great catch. Thanks @pkulikov

pkulikov commented on docs/csharp/tutorials/intro-to-csharp/arrays-and-collections.md:
```diff
@@ -1,19 +1,19 @@
 ms.custom: mvc
 ---
-# C# Quickstart: Collections
+# Introduction to C# tutorial: Collections
```
nit: should H1 header match the "title" in the metadata? 
I thought it's a guideline.

Thraka replied:
@pkulikov  No, the title should not match. Having different titles helps get this article pulled into more results.

pkulikov replied:
@Thraka cool! Good to know; thanks for the clarification.

pkulikov commented on docs/csharp/tutorials/intro-to-csharp/branches-and-loops-local.md:
```diff
@@ -1,18 +1,18 @@
 ---
-title: Branches and loops tutorial - C# local quickstarts
-description: In this quickstart about branches and loops, you write C# code to explore the language syntax that supports conditional branches and loops to execute statements repeatedly.
+title: Branches and loops tutorial - C# introduction tutorial
```
nit: duplication of the "tutorial" word and not consistent with the collections tutorial (which title ends with "Introduction to C#")

BillWagner replied:
Good catch. I changed all the titles to be **<subject> - Introduction to C# tutorial**, or **<subject> - Introduction to C# interactive tutorial**. The only exception is the string interpolation tutorial, which is not listed as "introductory".

pkulikov commented on docs/csharp/tutorials/intro-to-csharp/branches-and-loops-local.md:
```diff
@@ -1,18 +1,18 @@
 ms.date: 10/31/2017
 ms.custom: mvc
 ---
 # Branches and loops
```
one more nit about mismatch of H1 and the "title" in metadata

BillWagner replied:
per earlier comment, leaving these different.

pkulikov commented on docs/csharp/tutorials/intro-to-csharp/branches-and-loops.yml:
```diff
@@ -1,11 +1,10 @@
 metadata:
-  title:  Branches and loops interactive tutorial - C# quickstarts
-  description: In this quickstart about branches and loops, you'll use your browser to learn C# interactively. You'll write C# code and see the results of compiling and running your code directly in the browser.
+  title:  Branches and loops - Introduction to C# interactive tutorial
```
In one place "interactive" was replaced with "online". What about changing it everywhere: "interactive tutorial" -> "online tutorial"?

BillWagner replied:
per above comment, the titles are different than the H1s

pkulikov commented on docs/csharp/tutorials/intro-to-csharp/local-environment.md:
```diff
@@ -0,0 +1,62 @@
+description: This article provides a basic introduction to the tools you'll use to develop C# and .NET Applications on your machine.
+ms.date: 10/23/2018
+---
+# Become familiar with the .NET Development tools
```
nit: please lower-case "Development"

BillWagner committed:
Respond to feedback

- Fix redirection.
- Changed all the titles to be - Introduction to C# tutorial, or - Introduction to C# interactive tutorial. The only exception is the string interpolation tutorial, which is not listed as "introductory".
- update titles and H1s to provide different keywords.
- small grammar fixes.

BillWagner commented:
Ready for review @JennieHubbard

mairaw commented:
So what happens with the yml redirects? do you have to go through Akamai?
[This page](https://review.docs.microsoft.com/en-us/dotnet/csharp/tutorials/?branch=pr-en-us-8587) should also be revised, no?

mairaw commented:
Ah never mind my question about the yml. I see how they were done now.

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/interpolated-strings-local.md:
```diff
@@ -43,19 +42,19 @@ Let's try a few more string interpolation examples with some other data types.
 In the previous section, you used string interpolation to insert one string inside of another. The result of an interpolated expression can be of any data type, though. Let's include values of various data types in an interpolated string.
 
-In the following example, first, we define a [class](../programming-guide/classes-and-structs/classes.md) data type `Vegetable` that has the `Name` [property](../properties.md) and the `ToString` [method](../methods.md), which [overrides](../language-reference/keywords/override.md) the behavior of the <xref:System.Object.ToString?displayProperty=nameWithType> method. The [`public` access modifier ...
+In the following example, first, we define a [class](../../programming-guide/classes-and-structs/classes.md) data type `Vegetable` that has the `Name` [property](../../properties.md) and the `ToString` [method](../../methods.md), which [overrides](../../language-reference/keywords/override.md) the behavior of the <xref:System.Object.ToString?displayProperty=nameWithType> method. The [`public` acc ...
```
first, we define --> we first define
better "a Name property" and "a ToString method" instead of "the Name..."

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/interpolated-strings-local.md:
````diff
@@ -43,19 +42,19 @@ Let's try a few more string interpolation examples with some other data types.
 ```
 
-Then we create an instance of the `Vegetable` class by using [`new` keyword](../language-reference/keywords/new-operator.md) and providing a name parameter for the constructor `Vegetable`:
+Then we create an instance of the `Vegetable` class by using [`new` keyword](../../language-reference/keywords/new-operator.md) and providing a name parameter for the constructor `Vegetable`:
````
- mention here that the instance is named item; without it, line 58 is slightly mysterious
- using --> using the
- name parameter --> name argument, or just name

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/interpolated-strings.yml:
```diff
@@ -1,12 +1,11 @@
 ### YamlMime:Tutorial
-title: String interpolation in C#
+title: Use string interpolation to construct formatted strings 
```
a number of the comments from the previous article apply to this one as well.

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/interpolated-strings.yml:
```diff
@@ -15,7 +14,7 @@ metadata:
 - durationInMinutes: 2
   content: |
-    This quickstart teaches you how to use [string interpolation](../language-reference/tokens/interpolated.md) to insert values into a single string. You also learn how to control text representations of those values in the result string. You use your browser to write and run C# code and can immediately see the produced results.
+    This quickstart teaches you how to use [string interpolation](../../language-reference/tokens/interpolated.md) to insert values into a single string. You also learn how to control text representations of those values in the result string. You use your browser to write and run C# code and can immediately see the produced results.
```
produced results --> results

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/introduction-to-classes.md:
```diff
@@ -129,21 +129,21 @@ using System.Collections.Generic;
+[!code-csharp[DepositAndWithdrawal](../../../../samples/csharp/classes-quickstart/BankAccount.cs#DepositAndWithdrawal "Make deposits and withdrawals")]
 
-The [`throw`](../language-reference/keywords/throw.md) statement **throws** an exception. Execution of the current method ends, and will resume when a matching `catch` block is found. You'll add a `catch` block to test this code a little later on.
+The [`throw`](../../language-reference/keywords/throw.md) statement **throws** an exception. Execution of the current method ends, and will resume when a matching `catch` block is found. You'll add a `catch` block to test this code a little later on.
```
"execution of the current method ends" isn't accurate; it's only any remaining code in the throw block that isn't executed, and control flows to the catch block.

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/introduction-to-classes.md:
````diff
@@ -169,7 +169,7 @@ catch (ArgumentOutOfRangeException e)
 ```
 
-You use the [`try` and `catch` statements](../language-reference/keywords/try-catch.md) to mark a block of code that may throw exceptions, and to catch those errors that you expect. You can use the same technique to test the code that throws for a negative balance:
+You use the [`try` and `catch` statements](../../language-reference/keywords/try-catch.md) to mark a block of code that may throw exceptions, and to catch those errors that you expect. You can use the same technique to test the code that throws for a negative balance:
````
- no comma afer exception.
- for a novice audience, "throws an exception" is probably better than just "throws"

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/list-collection.yml:
```diff
@@ -28,9 +27,9 @@ items:
+    that you've learned in earlier tutorials to loop through the list.
 
-    The code to display names makes use of the [string interpolation](../language-reference/tokens/interpolated.md) feature. When you precede a `string` with the `$` character, you can embed C# code in the string declaration. The actual string replaces that C# code with the value it generates. In this example, it replaces the `{name.ToUpper()}` with each name, converted to capital letters, becaus ...
+    The code to display names makes use of the [string interpolation](../../language-reference/tokens/interpolated.md) feature. When you precede a `string` with the `$` character, you can embed C# code in the string declaration. The actual string replaces that C# code with the value it generates. In this example, it replaces the `{name.ToUpper()}` with each name, converted to capital letters, bec ...
```
the link to String.ToUpper should be ?displayProperty=nameWithType

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/local-environment.md:
```diff
@@ -0,0 +1,62 @@
+---
+# Become familiar with the .NET development tools
+
+The first step in running a tutorial on your machine is to setup a development environment.
```
setup --> set up

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/local-environment.md:
```diff
@@ -0,0 +1,62 @@
+
+In the [Numbers in C#](numbers-in-csharp-local.md) tutorial, you'll learn
+how computers store numbers and how to perform calculations with different
+numeric types. You'll learn the basics of rounding, and how to perform
```
no comma after "rounding"

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/local-environment.md:
```diff
@@ -0,0 +1,62 @@
+
+## [String interpolation](interpolated-strings-local.md)
+
+The [String interpolation](interpolated-strings-local.md) tutorial shows you how to insert values into a string. You'll learn how to create an interpolated string with embedded C# expressions and how to control the text appearance of the expression results in the result string.
```
text appearance --> formatting

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/numbers-in-csharp-local.md:
```diff
@@ -1,15 +1,14 @@
+# Manipulate integral and floating point number types in C# #
 
-# Numbers in C# quickstart
+This tutorial teaches you about the number types in C# interactively. You'll write small amounts of code, then you'll compile and run that code. The tutorial contains a series of lessons that explore numbers and math operations in C#. These lessons teach you the fundamentals of the C# language.
```
number types --> numeric types

rpetrusha commented on docs/csharp/tutorials/intro-to-csharp/numbers-in-csharp.yml:
```diff
@@ -15,7 +14,7 @@ metadata:
 - durationInMinutes: 1
   content: |
-    This quick start teaches you about the number types in C# interactively, using your browser to write C# and see the results of compiling and running your code. It contains a series of lessons that explore numbers and math operations in C#. These lessons teach you the fundamentals of the C# language.
+    This tutorial teaches you about the number types in C# interactively, using your browser to write C# and see the results of compiling and running your code. It contains a series of lessons that explore numbers and math operations in C#. These lessons teach you the fundamentals of the C# language.
```
number types --> numeric types

rpetrusha reviewed (approved):
This looks good, @BillWagner. I've left a number of suggestions for you to consider, though.

BillWagner committed:
respond to feedback.

BillWagner committed:
missed one build warning.
